### ISSUE TITLE
New and Edit Recipe Form

### ISSUE THREAD

**Issue by JayKay82** [JayKay82/recipe#16]
The form for creating recipes will be working with two controllers and models so I still need to figure out the best way to deal with this, especially because this form will grow in complexity with future iterations.

**Comment by felixclack** [comment]
Try reading https://robots.thoughtbot.com/activemodel-form-objects for an idea around handling the complexity of forms dealing with multiple objects.

Also, there's a library that can help too... https://github.com/apotonick/reform

**Comment by JayKay82** [comment]
I've implemented a form object using Reform. This is working fine. The only issue right now is with dynamically deleting nested ingredients with Javascript on updating a recipe. The solution I will attempt to implement is similar to create where I loop through post params and if a recipe has ingredients that have '_destroy' set to '1' it will delete it from the param. I will do the same with update only it will loop through params to check which ingredients were removed from the form (they will have '_destroy' set to '1') take the unique ingredient names and then match them to the reform object's ingredients array and destroy the necessary ingredients like that.
